pkg/breakglasscredential/flags.go

```diff
@@ -0,0 +1,144 @@
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	usernameFlag   = "username"
+	expirationFlag = "expiration"
+)
+
+type BreakGlassCredentialArgs struct {
+	username           string
+	expirationDuration time.Duration
+}
+
+func AddBreakGlassCredentialFlags(cmd *cobra.Command, prefix string) *BreakGlassCredentialArgs {
+	args := &BreakGlassCredentialArgs{}
+
+	cmd.Flags().StringVar(
+		&args.username,
+		usernameFlag,
+		"",
+		"Username for the break glass credential.",
+	)
+
+	cmd.Flags().DurationVar(
+		&args.expirationDuration,
+		expirationFlag,
+		0,
+		fmt.Sprintf("Expire the break glass credential after a relative duration like 2h, 8h. "+
+			"The expiration duration needs to be at least 10 minutes from now and to be at maximum 24 hours."),
+	)
+	return args
+}
+
+func GetBreakGlassCredentialOptions(cmd *pflag.FlagSet, prefix string,
+	confirmBeforeAllArgs bool, args *BreakGlassCredentialArgs) (
+	*BreakGlassCredentialArgs, error) {
+	var err error
+	result := &BreakGlassCredentialArgs{}
+
+	result.username = args.username
+	result.expirationDuration = args.expirationDuration
+
+	if interactive.Enabled() && !cmd.Changed(usernameFlag) {
+		result.username, err = interactive.GetString(interactive.Input{
+			Question: "Username",
+			Default:  result.username,
+			Help:     cmd.Lookup(usernameFlag).Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed(expirationFlag) {
+		inputString, err := interactive.GetString(interactive.Input{
+			Question: "Expiration duration",
+			Default:  result.expirationDuration,
+			Help:     cmd.Lookup(expirationFlag).Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if inputString != "" {
+			duration, err := time.ParseDuration(inputString)
+			if err != nil {
+				return nil, err
+			}
+			result.expirationDuration = duration
+		}
+	}
+
+	return result, nil
+}
+
+func CreateBreakGlass(cluster *cmv1.Cluster,
+	clusterKey string,
+	args *BreakGlassCredentialArgs, r *rosa.Runtime) error {
+
+	breakGlassConfig, err := CreateBreakGlassConfig(args)
+	if err != nil {
+		return fmt.Errorf("failed to create a break glass credential for cluster '%s': %s",
+			cluster.ID(), err)
+	}
+
+	_, err = r.OCMClient.CreateBreakGlassCredential(cluster.ID(), breakGlassConfig)
+
+	if err != nil {
+		return fmt.Errorf("failed to create a break glass credential for cluster '%s': %s",
+			cluster.ID(), err)
+	}
+	return nil
+
+}
+
+func CreateBreakGlassConfig(args *BreakGlassCredentialArgs) (*cmv1.BreakGlassCredential, error) {
+	breakGlassBuilder := cmv1.NewBreakGlassCredential()
+
+	if args.username != "" {
+		breakGlassBuilder.Username(args.username)
+	}
+
+	if args.expirationDuration != 0 {
+		expirationTimeStamp := time.Now().Add(args.expirationDuration).Round(time.Second)
+		breakGlassBuilder.ExpirationTimestamp(expirationTimeStamp)
+	}
+
+	breakGlassConfig, err := breakGlassBuilder.Build()
```

Comment:
@chenz4027 Can this be shortened to:

```
return breakGlassBuilder.Build()
```
?

Reply:
Will look into it & refactor the rest - Thanks @robpblake 👍

Reply:
Refactored